tlatools/org.lamport.tlatools/src/recomp/RecompVerify.java

```diff
@@ -47,151 +47,172 @@
-    	// model check the first component
-    	final String firstComp = components.get(0);
-		Utils.printVerbose(verbose, "");;
-		Utils.printVerbose(verbose, "Component 1" + ": " + firstComp);
-    	TLC tlcFirstComp = new TLC();
-    	timer.reset();
-    	tlcFirstComp.modelCheckOnlyGoodStates(firstComp, cfg); // TODO there's really no reason to distinguish between the 2 methods
-    	Utils.printVerbose(verbose, "State space gen: " + timer.timeElapsed() + "ms");
-    	Utils.assertNotNull(tlcFirstComp.getLTSBuilder(), "Error generating state space for the first component!");
-    	
-    	// turn the first component into a safety property (interface requirement)
-    	timer.reset();
-    	LTS<Integer, String> ltsProp = tlcFirstComp.getLTSBuilder().toIncompleteDetAutIncludingAnErrorState();
-    	Utils.printVerbose(verbose, "LTS gen: " + timer.timeElapsed() + "ms");
-    	Utils.printVerbose(verbose, "# unique states: " + (ltsProp.size()-1) + " states");
-    	int totalSumOfStatesChecked = ltsProp.size() - 1;
-    	int largestProductOfStatesChecked = ltsProp.size() - 1;
-    	//System.out.println();
-    	//FSPWriter.INSTANCE.write(System.out, ltsProp);
-    	//System.out.println();
-    	
-    	// minimize the LTS
-    	timer.reset();
-    	ltsProp = AutomataLibUtils.minimizeLTS(ltsProp);
-    	Utils.printVerbose(verbose, "minimization: " + timer.timeElapsed() + "ms");
-    	Utils.printVerbose(verbose, "# unique states post-minimization: " + (ltsProp.size()-1) + " states");
-    	
-    	// initialize the alphabet
-    	AlphabetMembershipTester alphabetTester = new AlphabetMembershipTester(tlcFirstComp.actionsInSpec(), ltsProp);
-    	
-    	if (SafetyUtils.INSTANCE.ltsIsSafe(ltsProp)) {
-    		final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
-    		Utils.printVerbose(verbose, "");
-    		System.out.println("k: " + 0);
-    		System.out.println("Total # states checked: " + totalNumStatesChecked);
-    		System.out.println("Property satisfied!");
-    		return;
-    	}
-    	if (SafetyUtils.INSTANCE.hasErrInitState(ltsProp)) {
-    		final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
-    		Utils.printVerbose(verbose, "");
-    		System.out.println("k: " + 0);
-    		System.out.println("Total # states checked: " + totalNumStatesChecked);
-			System.out.println("Property may be violated.");
-    		//FSPWriter.INSTANCE.write(System.out, ltsProp);
+		// write a config without any invariants / properties
+		final String noInvsCfg = "no_invs.cfg";
+		Utils.writeFile(noInvsCfg, "SPECIFICATION Spec");
+
+		try {
+			RVStrategy.runRVStrategies(tla, cfg, recompType, recompFile, verbose);
+		} catch (InterruptedException e) {
+			throw new RuntimeException(e);
+		}
+
+		System.exit(99);
+	}
+
+	public static void runRecompMap(final String tla, final String cfg, final String recompType, final String recompFile, boolean verbose, final List<String> rawComponents) {
```

Comment:
I would consider changing the method name form "runRecompMap" to "runRecompStrategy", since this method really runs a _strategy_, not just a map